app/components/post_draft/draft_input/index.tsx

```diff
@@ -133,16 +140,152 @@ export default function DraftInput({
     const intl = useIntl();
     const serverUrl = useServerUrl();
     const theme = useTheme();
+    const [isEmojiPickerOpen, setIsEmojiPickerOpen] = useState(false);
 
     const handleLayout = useCallback((e: LayoutChangeEvent) => {
         updatePostInputTop(e.nativeEvent.layout.height);
-    }, []);
+    }, [updatePostInputTop]);
+
+    // Use useRef to track and always retrieve the latest cursor position without triggering re-renders.
+    // This approach is used to improve performance, as the child component was re-rendering frequently
+    // due to isEmojiPress being added to the callback's dependency array.
+    // Further investigation may be needed to optimize this behavior.
+    const cursorPositionRef = useRef(cursorPosition);
+    useEffect(() => {
+        cursorPositionRef.current = cursorPosition;
+    }, [cursorPosition]);
+
+    const handleEmojiPress = useCallback((emojiName: string) => {
+        updateValue((v) => {
+            const name = emojiName.trim();
+            const currentCursorPosition = cursorPositionRef.current;
+            let unicode;
+            if (EmojiIndicesByAlias.get(name)) {
+                const emoji = Emojis[EmojiIndicesByAlias.get(name)!];
+                if (emoji.category === 'custom') {
+                    return `${v.slice(0, currentCursorPosition)} :${emojiName}: ${v.slice(currentCursorPosition)}`;
+                }
+                unicode = emoji.image;
+                if (unicode) {
+                    const codeArray = unicode.split('-'); // eslint-disable-next-line max-nested-callbacks
```

Comment:
Put the eslint disabled in a different line. This way looks like what you are disabling is the current line.